db/fixtures/miq_user_roles.yml

```diff
@@ -115,10 +115,7 @@
   - ems_cluster_perf
   - ems_cluster_tag
   - ems_cluster_timeline
-  - ems_infra_show
-  - ems_infra_show_list
-  - ems_infra_tag
-  - ems_infra_timeline
+  - ems_infra
```

Comment:
I can't tell why this was done.  Did you need to add another feature under `ems_infra` tree so instead of adding each one under the tree, we're adding the higher level `ems_infra` node?  

Is this for the "fix" part of the PR title?  I don't understand what was fixed with the information provided.

Reply:
The first 3 are Childs of the ems_infra so are included in ems_infra which is needed for the permission to work as expected.

Reply:
@ereztheodorou we need to know what wasn't working before and how this fixes it so that we can decide if there is a better way or if this is the right way to do it.

Reply:
Thanks.  Yeah, I guess what I'm asking is which specific product feature was missing? 


## Before
```
% grep -E "identifier.+ems_infra" db/fixtures/miq_product_features.yml
  :identifier: ems_infra
    :identifier: ems_infra_view
      :identifier: ems_infra_show_list     # +
      :identifier: ems_infra_show          # +
      :identifier: ems_infra_timeline      # +
    :identifier: ems_infra_control
      :identifier: ems_infra_discover
      :identifier: ems_infra_tag           # +          
      :identifier: ems_infra_protect
      :identifier: ems_infra_check_compliance
      :identifier: ems_infra_refresh
      :identifier: ems_infra_recheck_auth_status
      :identifier: ems_infra_import_vm
      :identifier: ems_infra_admin_ui
    :identifier: ems_infra_admin
      :identifier: ems_infra_delete
      :identifier: ems_infra_edit
      :identifier: ems_infra_new
      :identifier: ems_infra_scale
      :identifier: ems_infra_scaledown
      :identifier: ems_infra_change_password
  :identifier: ems_infra_dashboard
    :identifier: ems_infra_dashboard_view
```

## After
```
% grep -E "identifier.+ems_infra" db/fixtures/miq_product_features.yml
  :identifier: ems_infra                         # +
    :identifier: ems_infra_view                  # inherited from ems_infra
      :identifier: ems_infra_show_list           # inherited from ems_infra
      :identifier: ems_infra_show                # inherited from ems_infra
      :identifier: ems_infra_timeline            # inherited from ems_infra
    :identifier: ems_infra_control               # inherited from ems_infra
      :identifier: ems_infra_discover            # inherited from ems_infra
      :identifier: ems_infra_tag                 # inherited from ems_infra
      :identifier: ems_infra_protect             # inherited from ems_infra
      :identifier: ems_infra_check_compliance    # inherited from ems_infra
      :identifier: ems_infra_refresh             # inherited from ems_infra
      :identifier: ems_infra_recheck_auth_status # inherited from ems_infra
      :identifier: ems_infra_import_vm           # inherited from ems_infra
      :identifier: ems_infra_admin_ui            # inherited from ems_infra
    :identifier: ems_infra_admin                 # inherited from ems_infra
      :identifier: ems_infra_delete              # inherited from ems_infra
      :identifier: ems_infra_edit                # inherited from ems_infra
      :identifier: ems_infra_new                 # inherited from ems_infra
      :identifier: ems_infra_scale               # inherited from ems_infra
      :identifier: ems_infra_scaledown           # inherited from ems_infra
      :identifier: ems_infra_change_password     # inherited from ems_infra
  :identifier: ems_infra_dashboard               # inherited from ems_infra
    :identifier: ems_infra_dashboard_view        # inherited from ems_infra
```

So, clearly we're adding many new features to this user role and  my question is which features did you need and are we adding more features than are needed?

Reply:
What was broken that is being fixed here?

Reply:
This PR addresses 2 main issues: 
1. User was not able to add a volume as there were not enough permissions to allow it, so this was fixed here.
2. We had to add additional permissions to the physical user and admin ( "view only" for the user, and add/modify for admin) so the roles can be managed correctly by these personas.

Reply:
Should a physical storage administrator be able to change ems passwords?  Or delete ems's? Or scale them?  Or import vms? I don't know but it feels wrong.

I don't know which actual feature you need.  It's possible the calling code is checking the wrong product feature or we can add the one it needs.  Adding all of these features without actually needing them feels like a mistake.

If you aren't sure where the code is or what product feature it's checking, you can change the log level to debug in manageiq's settings/configuration advanced tab in the log level section which will look something like [this](https://github.com/ManageIQ/manageiq/blob/60be0beea46410e20409458287293a36169909ce/config/settings.yml#L857).  Change that to debug, save, and restart.  It should then log when product feature checks fail due to lack of permissions.  This will log in the evm.log/stdout on podified from either [here](https://github.com/ManageIQ/manageiq-ui-classic/blob/e36f61b78588851c65ed5cf8680e4fb68cb0a97d/app/helpers/application_helper.rb#L134) or [here](https://github.com/ManageIQ/manageiq/blob/60be0beea46410e20409458287293a36169909ce/lib/rbac/authorizer.rb#L34).

Reply:
@jrafanie Yes. A storage admin needs to be able to change the ems password add/delete/scale as this ems is managing the storage systems he is responsible for. This PR actually addresses the feedback we got on the previous roles PR from the storage team responsible for many of our client datacenters.

Reply:
@jrafanie If you need more info let me know.  We need it in for FP2 (morphy) before the cutoff so can we merge this?

Reply:
@jrafanie I checked with @felixnu and he tried to use the ems storage but it didn't work. The only thing that worked was the ems_infra so he had to assign those permissions. Lets get this PR in and open a issue for fixing the ems storage role to work.

Reply:
@felixnu can you open an issue describing specifically what didn't work?

Reply:
> @felixnu can you open an issue describing specifically what didn't work?

Yeah, I really think something else is wrong in the either the UI code, the provider, or something else.  I'd hate for us to "fix" the permission in the UI in 3 months and break this code because we relied on bad logic in the UI here.

Reply:
@agrare @jrafanie We can open the issue to address the problem , and link it to this PR  so we won't forget to change it once the problem is found and fixed in the UI/provider/something else.

Reply:
Please link the issue to this PR and we can move forward.  Thanks!

Reply:
@jrafanie created and linked.